docs/getting-started.md

```diff
@@ -202,19 +202,47 @@ $ENV:AWS_SECRET_ACCESS_KEY=$awsCredentials.SecretAccessKey
 
 ## Deploying a cluster
 
+### Starting Minikube
+For security reasons, it's best to start a Minikube instance before using
+clusterctl to store your AWS credentials.
+
+``` shell
+minikube start
+```
+
+Then write the AWS credentials into the Minikube instance for use by Cluster API:
+
+``` shell
+minikube ssh 'mkdir -p .aws'
+clusterawsadm alpha bootstrap generate-aws-default-profile | minikube ssh 'cat > .aws/credentials'
+[default]
+aws_access_key_id = AKIAIOSFODNN7EXAMPLE
+aws_secret_access_key = wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+region = eu-west-1
+
+
+^C
+```
+
+**NOTE**:
```

Comment:
Lets move this note above the codeblock to make it more obvious in advance.

Would it be better to use `minikube ssh-key` and standard `ssh` here instead to avoid the stdout issue?

Can we avoid this all together by moving to using a mounted secret as part of the deployment?

/cc @vincepri

Reply:
I started off with a mounted secret, but the problem is the pivot. Don't particularly want the users credentials in the pivoted cluster, and should defer to EC2 metadata (in either the AWS-provided or IAM metadata proxy form).

Reply:
From the notes above:

> During bootstrap, the user is instructed to use clusterawsadm to copy AWS credentials into Minikube under /home/docker/.aws/credentials which is then added as a
volume mount to the statefulset as /root/.aws.

> As no credentials are provided in the statefulset, the SDK will look for the shared credentials file, find it in the default path and load from there.

> After pivot, the new control plane will create an empty /home/docker/.aws directory, but because the user has run the CloudFormation bootstrap and our sample manifests specify the created IAM role, the controller will fall through to checking EC2 instance metadata and find credentials there to continue.

For reference, the Java SDK explains this best docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default

As far as I understand, if the secret isn't available on the target cluster, the pods won't start and clusterctl will fail, whereas the host volume path mount will create the directory if it doesn't exist?